### DISCUSSION TITLE
[Progress bar] access to total/completed

### DISCUSSION THREAD

**Discussion by andrzejnovak** [Textualize/rich#2058] [Q&A]
Is there a way to access the task total/completed with progress bar other than the private `_tasks` dict?

**Comment by jfthuong** [comment]
Yes, you can just add a `MofNCompleteColumn`: https://rich.readthedocs.io/en/latest/reference/progress.html#rich.progress.MofNCompleteColumn

The documentation explains how to do that here:
https://rich.readthedocs.io/en/latest/progress.html#columns

**Reply by andrzejnovak** [reply]
That's not the question though. I need to modify the total at runtime, currently using `progress._tasks[pid].total + 1` but would prefer to not do that over private API.

**Comment by jfthuong** [comment]
The question was not clear: you mention `total/completed`, which sounds like the display of these two. And you just mention `access`.

You could probably improve the initial question with your use case.

For displaying: you have the previous answer.

For writing: just look at the documentation of the function `Progress.update` https://rich.readthedocs.io/en/latest/reference/progress.html#rich.progress.Progress.update
